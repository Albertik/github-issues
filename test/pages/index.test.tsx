import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import App from '../../pages/index';

describe('App', () => {
	afterEach(cleanup);

	it('renders without crashing', () => {
		const { asFragment } = render(<App />);

		expect(
			screen.getByRole('heading', { name: 'Hi, APC 👋' })
		).toBeInTheDocument();

		expect(asFragment()).toMatchSnapshot();
	});

	it('should link to /searchOrganizationRepositories', () => {
		render(<App />);

		expect(screen.getByTestId('linkText').textContent).toBe(
			"Search organization's repositories 🕵️‍♂️"
		);
	});
});
