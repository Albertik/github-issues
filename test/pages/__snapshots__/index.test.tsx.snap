// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="css-1bjqwbw"
  >
    <main
      class="css-n1ih9h"
    >
      <h1
        class="css-ih13jk"
      >
        Hi, APC 👋
      </h1>
      <a
        class="css-12ol0na"
        data-testid="linkText"
      >
        Search organization's repositories 🕵️‍♂️
      </a>
    </main>
  </div>
</DocumentFragment>
`;
